BuildMacOSUniveralBinary.py

```diff
@@ -0,0 +1,146 @@
+"""
+The current tooling supported in CMake, Homebrew, and QT5 are insufficient for creating
+MacOSX universal binaries automatically for applications like Dolphin which have more 
+complicated build requirements (like different libraries, build flags and source files
+for each target architecture). 
+
+So instead, this script manages the conifiguration and compilation of distinct builds
+and project files for each target architecture and then merges the two binaries into
+a single universal binary. 
+
+Running this script will:
+1) Generate Xcode project files for the ARM build (if project files don't already exist)
+2) Generate Xcode project files for the x64 build (if project files don't already exist)
+3) Build the ARM project for the selected build_target
+4) Build the x64 project for the selected build_target
+5) Generates universal .app packages combining the ARM and x64 packages
+6) Utilizes the lipo tool to combine the binary objects inside each of the packages
+   into universal binaries
+7) Code signs the final universal binaries using the specified codesign_identity
+"""
+
+##BEGIN CONFIG##
+#Location of destination universal binary
+dst_app = "universal/"
+#Build Target (dolphin-emu to just build the emulator and skip the tests)
+build_target = "ALL_BUILD"
+
+#Locations to pkg config files for arm and x64 libraries
+#The default values of these paths are taken from the default
+#paths used for homebrew 
+arm_pkg_config_path='/opt/homebrew/lib/pkgconfig'
+x64_pkg_config_path='/usr/local/lib/pkgconfig'
```

Comment:
I wonder if those (and perhaps other) paths should be configurable (or auto-detectable? if possible?) in a way. Modifying files that are versioned just to build something seems off (not just in CI but also locally)

Reply:
These paths are now configurable through command line options in the script.